### DISCUSSION TITLE
IPv6Address won't get removed

### DISCUSSION THREAD

**Discussion by docbobo** [projectcalico/calico#9621] [Q&A]
A few weeks ago I tried enabling ipv6 on my network but had to disable that again due to some weird issues. Ever since, I get the following log outputs in `calico-node` over and over again:

```
bird: Mesh_2a02_8108_4c8d_3b01_a022_2eff_febd_789c: Socket error: bind: Cannot assign requested address
```

I believe this is due to the fact that there is `projectcalico.org/IPv6Address` annotation on each of my nodes, but I have no idea how to get rid of it. As far as I am concerned, I have removed all the settings that would enable ipv6 support, the nodes do not receive an ipv6 address either. 

I tried manually removing the annotation, but that resulted in `calico-node` failing to start at all.

This is on 3.29.1

**Comment by caseydavenport** [comment]
@docbobo could you share how you've installed Calico and how you enabled / disabled IPv6 in this case? Is this a Calico operator install or something else? If so, could you share your `Installation` resource?

Might be worth raising an issue and filling out the template, which prompts for some of the information that we'll need to diagnose: https://github.com/projectcalico/calico/issues/new/choose

**Comment by 13567436138** [comment]
I have the same problem

**Comment by docbobo** [comment]
@caseydavenport first of all, me need to disable this went away. But maybe that piece of information is helpful as well.

I've installed Calico via the operator, then followed the instructions to enable ipv6 dual stack. The reason I wanted to disable ipv6 again was due to the fact that my nodes were losing their ipv6 addresses after a few hours. As this results in the bird error mentioned above, I wanted to disable ipv6 again.

Now, in the meantime I found out that for some weird reason my nodes were not taking into account the default for `net.ipv6.conf.*.accept_ra` and  once I got that fixed, I have stable ipv6 addresses for my network interfaces and thus also no need to disable this anymore.

**Comment by caseydavenport** [comment]
Looks like right now, the code keeps the old value even if IPv6 is disabled: https://github.com/projectcalico/calico/blob/f7d97642b0dcd7f75223967419f16ac193bd9d1a/node/pkg/lifecycle/startup/startup.go#L559-L561

Probably need a way for the operator to tell calico/node to clear any configured IPv6 address.

> I tried manually removing the annotation, but that resulted in calico-node failing to start at all.

This is a little bit surprising to me. Do you have the error log from this by chance?

**Comment by docbobo** [comment]
When I manually remove the  `projectcalico.org/IPv6Address` annotation, the following happens when calico-node comes up:

```
calico-node 2025-01-22 12:58:13.168 [INFO][9] startup/startup.go 439: Early log level set to info                                                                                                                                                                                                                        
calico-node 2025-01-22 12:58:13.169 [INFO][9] startup/utils.go 126: Using NODENAME environment for node name talos-tp1-n02                                                                                                                                                                                               
calico-node 2025-01-22 12:58:13.169 [INFO][9] startup/utils.go 138: Determined node name: talos-tp1-n02                                                                                                                                                                                                                  
calico-node 2025-01-22 12:58:13.169 [INFO][9] startup/startup.go 95: Starting node talos-tp1-n02 with version v3.29.1                                                                                                                                                                                                    
calico-node 2025-01-22 12:58:13.170 [INFO][9] startup/startup.go 444: Checking datastore connection                                                                                                                                                                                                                      
calico-node 2025-01-22 12:58:13.180 [INFO][9] startup/startup.go 468: Datastore connection verified                                                                                                                                                                                                                      
calico-node 2025-01-22 12:58:13.180 [INFO][9] startup/startup.go 105: Datastore is ready                                                                                                                                                                                                                                 
calico-node 2025-01-22 12:58:13.191 [WARNING][9] startup/winutils.go 150: Neither --kubeconfig nor --master was specified.  Using the inClusterConfig.  This might not work.                                                                                                                                             
calico-node 2025-01-22 12:58:13.210 [INFO][9] startup/autodetection_methods.go 103: Using autodetected IPv4 address on interface end0: 192.168.10.12/24                                                                                                                                                                  
calico-node 2025-01-22 12:58:13.212 [WARNING][9] startup/autodetection_methods.go 99: Unable to auto-detect an IPv6 address: no valid IPv6 addresses found on the host interfaces                                                                                                                                        
calico-node 2025-01-22 12:58:13.212 [WARNING][9] startup/startup.go 552: Couldn't autodetect an IPv6 address. If auto-detecting, choose a different autodetection method. Otherwise provide an explicit address.                                                                                                         calico-node 2025-01-22 12:58:13.212 [INFO][9] startup/startup.go 403: Clearing out-of-date IPv4 address from this node IP="192.168.10.12/24"                                                                                                                                                                             
calico-node 2025-01-22 12:58:13.212 [INFO][9] startup/startup.go 407: Clearing out-of-date IPv6 address from this node IP=""                                                                                                                                                                                             
calico-node 2025-01-22 12:58:13.231 [WARNING][9] startup/utils.go 48: Terminating                                                                                                                                                                                                                                        
calico-node Calico node failed to start
```

Let me know if you need anything else.

**Comment by caseydavenport** [comment]
> calico-node 2025-01-22 12:58:13.212 [WARNING][9] startup/autodetection_methods.go 99: Unable to auto-detect an IPv6 address: no valid IPv6 addresses found on the host interfaces                                                                                                                                        

Sounds like Calico is still configured to use IPv6 autodetection? Do you have this `spec.calicoNetwork.nodeAddressAutodetectionV6` field configured?

https://github.com/tigera/operator/blob/master/api/v1/installation_types.go#L516-L519
